### DISCUSSION TITLE
How to get the Counters of IPSec tunnel?

### DISCUSSION THREAD

**Discussion by tyaguvg** [strongswan/strongswan#1340] [Q&A]
Hi,

Can someone help me of how I can get the counters of ipsec traffic transaction as listed below?

counters like;

!
in-packets
in-bytes rate
in-bytes
out-packets
out-bytes rate
out-bytes
errors
!

Thank you,
tyagu

**Comment by Thermi** [comment]
in-packets <- attribute of the SA
in-bytes rate <- not provided
in-bytes <- attribute of the SA
out-packets <- attribute of the SA
out-bytes rate <- not provided
out-bytes <- attribute of the SA
errors <- only global counter available (AFAIK) /proc/net/xfrm_stat

**Comment by tyaguvg** [comment]
Thank you.

Can you please help me with the commands that I can use to get the details
(whatever is provided)?

**Comment by tobiasbrunner** [comment]
You might want to look into [RADIUS Accounting](https://docs.strongswan.org/docs/5.9/plugins/eap-radius.html#_accounting) as that allows collecting data across rekeyings.

**Reply by cyanideangel2000** [reply]
If you are looking for simple blunt numbers, swanctl --list-sas does give us the bytes and packets counters per child_sa to judge the rx tx rate.

**Reply by tyaguvg** [reply]
> If you are looking for simple blunt numbers, swanctl --list-sas does give us the bytes and packets counters per child_sa to judge the rx tx rate.

Thank you Varun.

I'm trying that command but getting error as shown below;

root@ubuntu:~# swanctl --list-sas
connecting to 'unix:///var/run/charon.vici' failed: No such file or directory
Error: connecting to 'default' URI failed: No such file or directory
strongSwan 5.8.2 swanctl
usage:
  swanctl --list-sas [--raw|--pretty]
           --help            (-h)  show usage information
           --ike             (-i)  filter IKE_SAs by name
           --ike-id          (-I)  filter IKE_SAs by unique identifier
           --noblock         (-n)  don't wait for IKE_SAs in use
           --raw             (-r)  dump raw response message
           --pretty          (-P)  dump raw response message in pretty print
           --debug           (-v)  set debug level, default: 1
           --options         (-+)  read command line options from file
           --uri             (-u)  service URI to connect to
root@ubuntu:~#
root@ubuntu:~# swanctl --list-sas --ike
swanctl: option '--ike' requires an argument
Error: invalid options
strongSwan 5.8.2 swanctl
usage:
  swanctl --list-sas [--raw|--pretty]
           --help            (-h)  show usage information
           --ike             (-i)  filter IKE_SAs by name
           --ike-id          (-I)  filter IKE_SAs by unique identifier
           --noblock         (-n)  don't wait for IKE_SAs in use
           --raw             (-r)  dump raw response message
           --pretty          (-P)  dump raw response message in pretty print
           --debug           (-v)  set debug level, default: 1
           --options         (-+)  read command line options from file
           --uri             (-u)  service URI to connect to
root@ubuntu:~# swanctl --list-sas --raw
connecting to 'unix:///var/run/charon.vici' failed: No such file or directory
Error: connecting to 'default' URI failed: No such file or directory
strongSwan 5.8.2 swanctl
usage:
  swanctl --list-sas [--raw|--pretty]
           --help            (-h)  show usage information
           --ike             (-i)  filter IKE_SAs by name
           --ike-id          (-I)  filter IKE_SAs by unique identifier
           --noblock         (-n)  don't wait for IKE_SAs in use
           --raw             (-r)  dump raw response message
           --pretty          (-P)  dump raw response message in pretty print
           --debug           (-v)  set debug level, default: 1
           --options         (-+)  read command line options from file
           --uri             (-u)  service URI to connect to
root@ubuntu:~# swanctl --list-sas --r
connecting to 'unix:///var/run/charon.vici' failed: No such file or directory
Error: connecting to 'default' URI failed: No such file or directory
strongSwan 5.8.2 swanctl
usage:
  swanctl --list-sas [--raw|--pretty]
           --help            (-h)  show usage information
           --ike             (-i)  filter IKE_SAs by name
           --ike-id          (-I)  filter IKE_SAs by unique identifier
           --noblock         (-n)  don't wait for IKE_SAs in use
           --raw             (-r)  dump raw response message
           --pretty          (-P)  dump raw response message in pretty print
           --debug           (-v)  set debug level, default: 1
           --options         (-+)  read command line options from file
           --uri             (-u)  service URI to connect to
root@ubuntu:~#


Thank you
